### PR TITLE
 Issue #139 - Application sync should delete 'unexpected' resources

### PR DESCRIPTION
**Pull request by alexmt**
Closes https://github.com/argoproj/argo-cd/issues/139

### CODE REVIEW

File: server/application/application.go [hunk 1]
```diff
@@ -517,41 +513,85 @@ func (s *Server) deploy(
...
-		needsCreate := bool(liveObjs[i] == nil)
-		if !diffRes.Modified {
+
+		if resourceState.Status == appv1.ComparisonStatusSynced {
 			resDetails.Message = fmt.Sprintf("already synced")
 		} else if dryRun {
 			if needsCreate {
 				resDetails.Message = fmt.Sprintf("will create")
+			} else if needsDelete {
+				resDetails.Message = fmt.Sprintf("will delete")
```
Comment (jessesuen) [inline server/application/application.go]: This needs to take consideration the prune flag. Can we use the term "ignored" if we decide not to delete the resource?

File: server/application/application.go [hunk 2]
```diff
@@ -517,41 +513,85 @@ func (s *Server) deploy(
...
+			if needsDelete {
+				if prune {
+					err = kube.DeleteResource(config, liveObj, namespace)
+					if err != nil {
+						return nil, nil, err
+					}
+
+					resDetails.Message = fmt.Sprintf("deleted")
+				} else {
+					resDetails.Message = fmt.Sprintf("skipped (should be deleted)")
```
Comment (jessesuen) [inline server/application/application.go]: Lets use the term "ignored"

**Review by jessesuen — CHANGES_REQUESTED** [review]

**Review by merenbach — APPROVED** [review]

File: server/application/application.go (hunk 1 above)
Comment (alexmt) [reply server/application/application.go]: done

File: server/application/application.go (hunk 2 above)
Comment (alexmt) [reply server/application/application.go]: done

**Comment by alexmt** [comment]
@jessesuen your comments are implemented. PTAL

**Review by jessesuen — APPROVED** [review]

**Pull request merged**